bluemira/equilibria/optimisation/problem/base.py

```diff
@@ -75,6 +83,44 @@ class CoilsetOptimisationProblem(abc.ABC):
     to a specific objective function for that subclass.
     """
 
+    def __init__(
+        self,
+        coilset: CoilSet,
+        opt_algorithm: AlgorithmType,
+        *,
+        max_currents: npt.ArrayLike | None = None,
+        targets: MagneticConstraintSet | None = None,
+        constraints: list[UpdateableConstraint] | None = None,
+        opt_conditions: dict[str, float | int] | None = None,
+        opt_parameters: dict[str, float] | None = None,
+    ):
+        self._coilset = coilset
+        self.max_currents = max_currents
+        self.bounds = self.get_current_bounds(self.coilset, max_currents, self.scale)
+
+        self.targets = targets or MagneticConstraintSet([])
+        self.constraints = constraints or []
+
+        self.opt_algorithm = opt_algorithm
+        self.opt_conditions = self._opt_condition_defaults(
+            {"max_eval": 100} if opt_conditions is None else opt_conditions
```

Comment:
This is possibly a wider problem (that was already there) but if you specify anything in opt_conditions the max_eval (and same with step size elsewhere) wont be fed through.

maybe just this to fix
```
{"max_eval": 100} if opt_conditions is None else {"max_eval": 100, **opt_conditions}
```
or 
```
opt_conditions = {"max_eval": 100, **(opt_conditions or {})}
```